test/BUILD.bazel

```diff
@@ -1,54 +1,263 @@
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "ctor_dtor",
+    srcs = ["test_ctor_dtor.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "dump",
+    srcs = ["test_dump.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "enum",
+    srcs = [
+        "test_enum.cpp",
+        "test_enum_common.hpp",
+    ],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "error_exception",
+    srcs = ["test_error_exception.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
         "//:c4core",
-        "@doctest//doctest:custom_main",
     ],
 )
 
-[cc_test(
-    name = test,
+cc_test(
+    name = "error",
+    srcs = ["test_error.cpp"],
     includes = ["."],
-    srcs = ["test_%s.cpp" % test] + _tests[test],
-    deps = [":lib", "//:c4core"],
-) for test in _tests]
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "format",
+    srcs = ["test_format.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "memory_util",
+    srcs = ["test_memory_util.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "preprocessor",
+    srcs = ["test_preprocessor.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "span",
+    srcs = ["test_span.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "std_string",
+    srcs = ["test_std_string.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "std_vector",
+    srcs = ["test_std_vector.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "substr",
+    srcs = ["test_substr.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "type_name",
+    srcs = ["test_type_name.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "types",
+    srcs = ["test_types.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "utf",
+    srcs = [
+        "test_utf.cpp",
+        "utfchars.inc",
+    ],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "memory_resource",
+    srcs = ["test_memory_resource.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
+
+cc_test(
+    name = "szconv",
+    srcs = ["test_szconv.cpp"],
+    includes = ["."],
+    deps = [
+        ":lib",
+        "//:c4core",
+    ],
+)
```

Comment:
@zaucy now that the tests succeed, and given that this will have to be maintained in the future, I'd like to understand the reasoning behind this commit and how it enables the NaN test to succeed.

Reply:
I'm pretty sure it didn't fix it. https://github.com/zaucy/c4core/actions/runs/2908604060 At least on my instance of GitHub actions it's still the same failure. Just the linux ones got farther along now.